src/engine/runtime.js

```diff
@@ -552,6 +570,7 @@ Runtime.prototype._step = function () {
     this.redrawRequested = false;
     var inactiveThreads = this.sequencer.stepThreads();
     this._updateGlows(inactiveThreads);
+    this._setThreadCount(this.threads.length);
```

Comment:
`doneThreads` seems like a good name!

It is possible that `stepThreads` will cause new threads to be "generated". For example, when you run a "broadcast" block, it could start one new thread for each when-broadcast hat.